src/cryptography/x509.py

```diff
@@ -237,6 +237,40 @@ def __repr__(self):
                 "path_length={0.path_length})>").format(self)
 
 
+class KeyUsage(object):
+    def __init__(self, digital_signature, non_repudiation, key_encipherment,
+                 data_encipherment, key_agreement, key_cert_sign, crl_sign,
+                 encipher_only, decipher_only):
+        if not key_agreement and (encipher_only or decipher_only):
+            raise ValueError(
+                "encipher_only and decipher_only can only be true when "
+                "key_agreement is true"
+            )
+
+        self._digital_signature = digital_signature
+        self._non_repudiation = non_repudiation
+        self._key_encipherment = key_encipherment
+        self._data_encipherment = data_encipherment
+        self._key_agreement = key_agreement
+        self._key_cert_sign = key_cert_sign
+        self._crl_sign = crl_sign
+        self._encipher_only = encipher_only
+        self._decipher_only = decipher_only
+
+    digital_signature = utils.read_only_property("_digital_signature")
+    non_repudiation = utils.read_only_property("_non_repudiation")
+    # Later versions of X509 have renamed non_repudiation to
+    # content_commitment
```

Comment:
What's difficult about it? You write a computer program, you try to find
the name, you search the docs and it says "This used to be called blah blah
blah. under the new name. It's not like we have legacy code to support.

On Tue, Mar 31, 2015 at 8:55 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/x509.py
> <https://github.com/pyca/cryptography/pull/1802#discussion_r27537415>:
>
> > +            )
> > +
> > +        self._digital_signature = digital_signature
> > +        self._non_repudiation = non_repudiation
> > +        self._key_encipherment = key_encipherment
> > +        self._data_encipherment = data_encipherment
> > +        self._key_agreement = key_agreement
> > +        self._key_cert_sign = key_cert_sign
> > +        self._crl_sign = crl_sign
> > +        self._encipher_only = encipher_only
> > +        self._decipher_only = decipher_only
> > +
> > +    digital_signature = utils.read_only_property("_digital_signature")
> > +    non_repudiation = utils.read_only_property("_non_repudiation")
> > +    # Later versions of X509 have renamed non_repudiation to
> > +    # content_commitment
>
> While content_commitment is the "new" name, everyone still knows it by
> the old one so it'd be difficult to not support it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1802/files#r27537415>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084